config/config.go

```diff
@@ -268,6 +276,25 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	return includedConfig, nil
 }
 
+// Merge the extra arguments prioritizing those defined in the childExtraArgs
+func mergeExtraArgs(terragruntOptions *options.TerragruntOptions, childExtraArgs []TerraformExtraArguments, parentExtraArgs *[]TerraformExtraArguments) {
+	result := *parentExtraArgs
+addExtra:
+	for _, child := range childExtraArgs {
+		for i, parent := range result {
+			if parent.Name == child.Name {
+				terragruntOptions.Logger.Printf("extra_arguments '%v' from child overriding parent", child.Name)
+				// For extra args, we want to keep the values specified in the child and put them after
+				// the parent ones, so if we encounter a duplicate, we just overwrite it.
+				result[i] = child
+				continue addExtra
```

Comment:
Totally understood @jocgir .   Makes sense given go's limitations.

@brikis98 I swapped out the continue/label with a slightly different implementation that I think is a bit more clear.  Let me know if this works for you.